platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -240,7 +240,11 @@ class DefaultConfigurationCache internal constructor(
             // Can reuse the cache entry for the rest of this build invocation
             cacheAction = ConfigurationCacheAction.LOAD
         }
-        scopeRegistryListener.dispose()
+        try {
```

Comment:
I think this could use some contextual information so the next reader understands why it is important that that we stop the fingerprint controller before disposing the scope registry listener, and that we dispose the scope registry listener even if the CFC fails to stop.